### PR TITLE
Added tests for juniper_junos_rpc

### PR DESCRIPTION
**Pull request by rsmekala**
I have added the test in Jenkins as well.

Thanks to the tests. I was able to find some minor bugs in `juniper_junos_common` 

- In case we are not able to convert it to bool, we should return `None`. Currently, it causes a `TypeError`.
- The mode should be `ab` instead of `a` to allow python3 to write bytestrings

### CODE REVIEW

**Comment by rsmekala** [comment]
ok to test

**Comment by rsmekala** [comment]
retest this please

File: module_utils/juniper_junos_common.py [hunk 1]
```diff
@@ -1173,7 +1173,11 @@ def parse_arg_to_list_of_dicts(self,
                 if allow_bool_values is True:
                     # Try to convert it to a boolean value. Will be None if it
                     # can't be converted.
-                    bool_val = boolean(v)
+                    try:
+                        bool_val = boolean(v)
+                    except TypeError:
+                        bool_val = None
+                        pass
```
Comment (vnitinv) [inline module_utils/juniper_junos_common.py]: remove this line

**Comment by rsmekala** [comment]
retest this please

File: module_utils/juniper_junos_common.py (hunk 1 above)
Comment (rsmekala) [reply module_utils/juniper_junos_common.py]: Fixed

**Pull request merged**